src/tests/datasets.h

```diff
@@ -4,74 +4,31 @@
 #include <string>
 #include <vector>
 
-static const auto test_data_dir = std::filesystem::current_path() / "input_data";
+#include "config/tabular_data/input_table_type.h"
+#include "parser/csv_parser/csv_parser.h"
 
-struct Dataset {
-    std::string name;
-    size_t hash;
+static auto const test_data_dir = std::filesystem::current_path() / "input_data";
+
+namespace tests {
+
+/// dataset configuration info to create an input table
+struct DatasetInfo {
+    std::string_view name;
```

Comment:
I think we should use `std::string` here, since this data member actually _owns_ the data it holds, so it shouldn't be `view` type. And by using `string_view` here, we're basically restricting ourselves to use only string literals for dataset names in the future, which doesn't seem like a right thing